src/components/course-listing.cjsx

```diff
@@ -60,12 +60,19 @@ CourseListing = React.createClass
       isStudent = _.find roles, (role) -> role.type is 'student'
       isTeacher = _.find roles, (role) -> role.type is 'teacher'
 
+      courseType = if isConceptCoach then 'Concept Coach' else 'Tutor'
+
+      if courselink?
+        courseTypeFlag = "(#{courseType})"
+      else
+        courseTypeFlag = <div className='course-type-flag'>{courseType}</div>
+
```

Comment:
It seems to me that something weird's going on with the `courseLink` var.  It's not assigned until way below in the else statement. Or am I missing something?  I wonder if that's an artifact from how the listing used to work before we had archived periods to worry about?  Seems like this entire block could be re-written to be a bit clearer.